zerver/views/thumbnail.py

```diff
@@ -39,7 +40,14 @@ def backend_serve_thumbnail(
         assert isinstance(maybe_user_profile, UserProfile)
         realm = maybe_user_profile.realm
 
-    if not validate_thumbnail_request(realm, maybe_user_profile, url):
+    is_authorized = validate_thumbnail_request(realm, maybe_user_profile, url)
```

Comment:
As I [noted on CZO](https://chat.zulip.org/#narrow/stream/3-backend/topic/404s.20for.20deleted.20files.20.2323739/near/1520743), in development the thumbnailing system is still enabled -- but it does nothing, and this serves to confuse things a great deal, because it means you're not actually testing the same codepaths which are used in production.

So making a change here isn't _wrong_, but it's not going to address the issue for all real deploys.  We need to make a change here, but more importantly, we need to also make the `zerver.views.upload` codepath use the same logic (i.e. being sensitive to the `Accept` header).  You should edit (but not check in!) `zproject/dev_settings.py` to set `THUMBNAIL_IMAGES = False` to see the other codepath in action.